django/contrib/gis/db/models/functions.py

```diff
@@ -0,0 +1,397 @@
+    pass
+
+
+class Length(DistanceResultMixin, GeoFunc):
+    output_field_class = FloatField
+
+    def __init__(self, expr1, spheroid=True, **extra):
+        self.spheroid = spheroid
+        super(Length, self).__init__(expr1, **extra)
+
+    def as_sql(self, compiler, connection):
+        geo_field = GeometryField(srid=self.srid)  # Fake field to get SRID info
+        if geo_field.geodetic(connection) and not connection.features.supports_length_geodetic:
+            raise NotImplementedError("This backend doesn't support Length on geodetic fields")
+        return super(Length, self).as_sql(compiler, connection)
+
+    def as_postgresql(self, compiler, connection):
+        geo_field = GeometryField(srid=self.srid)  # Fake field to get SRID info
+        src_field = self.get_source_fields()[0]
+        geography = src_field.geography and self.srid == 4326
+        if geography:
+            self.source_expressions.append(Value(self.spheroid))
+        elif geo_field.geodetic(connection):
+            # Geometry fields with geodetic (lon/lat) coordinates need length_spheroid
+            self.function = 'ST_Length_Spheroid'
+            self.source_expressions.append(Value(geo_field._spheroid))
+        else:
+            dim = min(f.dim for f in self.get_source_fields() if f)
+            if dim > 2:
+                self.function = connection.ops.length3d
+        return super(Length, self).as_sql(compiler, connection)
+
+    def as_sqlite(self, compiler, connection):
+        geo_field = GeometryField(srid=self.srid)
+        if geo_field.geodetic(connection):
+            if self.spheroid:
+                self.function = 'GeodesicLength'
+            else:
+                self.function = 'GreatCircleLength'
+        return super(Length, self).as_sql(compiler, connection)
+
+
+class MemSize(GeoFunc):
+    output_field_class = IntegerField
+
+
+class NumGeometries(GeoFunc):
+    output_field_class = IntegerField
+
+
+class NumPoints(GeoFunc):
+    output_field_class = IntegerField
+
+    def as_sqlite(self, compiler, connection):
+        if self.source_expressions[self.geom_param_pos].output_field.geom_type != 'LINESTRING':
+            raise TypeError("Spatialite NumPoints can only operate on LineString content")
+        return super(NumPoints, self).as_sql(compiler, connection)
+
+
+class Perimeter(DistanceResultMixin, GeoFunc):
+    output_field_class = FloatField
+
+    def as_postgresql(self, compiler, connection):
+        dim = min([f.dim for f in self.get_source_fields()])
+        if dim > 2:
+            self.function = connection.ops.perimeter3d
+        return super(Perimeter, self).as_sql(compiler, connection)
+
+
+class PointOnSurface(GeoFunc):
+    pass
+
+
+class Reverse(GeoFunc):
+    pass
+
+
+class Scale(GeoFunc):
+    def __init__(self, expression, x, y, z=0.0, **extra):
+        expressions = [
+            expression,
+            self._handle_param(x, 'x', six.integer_types + (float,)),
```

Comment:
module level constant for `six.integer_types + (float,)` to avoid repeating it often?